### ISSUE TITLE
markdownlint.configFile absolute path is incorrectly resolved when workspace opened via UNC path

### ISSUE THREAD

**Issue by Marble-Palette** [DavidAnson/vscode-markdownlint#457]
## Environment

- VS Code: 1.131.0
- vscode-markdownlint: 0.62.1
- Windows

## Configuration

```json
{
    "markdownlint.configFile": "C:\\temp\\.markdownlint.yml"
}
```

## Repro steps

1. Open a Markdown file from a UNC workspace:
   `\\server\share\folder`

2. Open a Markdown file in that workspace.

## Expected behavior

The extension loads the configured file:

```text
C:\temp\.markdownlint.yml
```

## Actual behavior

The extension reports:

```text
Error: Unable to use configuration file
'C:/temp/.markdownlint.yml';

Error: UNKNOWN: unknown error, open
'\\server\.markdownlint.yml'
```

## Additional observations

- Local workspaces work correctly.
- A mapped network drive (for example `Z:`) works correctly.
- The issue only occurs when the workspace is opened using a UNC path.
- The problem occurs regardless of where the config file is located.
- Tested with:
  - `C:\Users\<user>\.markdownlint.yml`
  - `C:\temp\.markdownlint.yml`
- The configured absolute path appears to be replaced during resolution when the workspace root is a UNC path.

It looks like either vscode-markdownlint or markdownlint-cli2 is incorrectly resolving the configured `configFile` path when the workspace URI uses a UNC path.

## Result matrix

Local workspace : OK
Mapped drive    : OK
UNC workspace   : NG

**Comment by DavidAnson** [comment]
Sorry, Windows path handling is always challenging. It seems you have a workaround which is good. The documentation recommends forward slash everywhere, so you might try that. You could also try `${userHome}/file`. And if you experiment with CLI2 directly, please share your findings. Thank you!

**Comment by Marble-Palette** [comment]
Thanks for the suggestions! I did some additional investigation.

Unfortunately, the issue still occurs when the workspace is opened via a UNC path, even with the `${userHome}` syntax using forward slashes:

```json
{
    "markdownlint.configFile": "${userHome}/dotfiles/.markdownlint.yaml"
}
```

It appears that `fwFolderUriWithPathSegment` is being applied to the config file's absolute path, rewriting it from  `C:/Users/<user>/dotfiles/.markdownlint.yaml` to `file://<server>/<user>/dotfiles/.markdownlint.yaml`.

```javascript
fwFolderUriWithPathSegment(e) {
    let t = pd(e);
    // strip drive letter "C:"
    // `C:/Users/<user>/dotfiles/.markdownlint.yaml` to `/Users/<user>/dotfiles/.markdownlint.yaml`
    return Jp.test(t) && (t = this.fwFolderUri.path.startsWith("/") && Jp.test(this.fwFolderUri.path.slice(1)) ? `/${t}` : t.replace(Jp, "/")), 
    // strip first segment "/Users"
    // `/Users/<user>/dotfiles/.markdownlint.yaml` to `/<user>/dotfiles/.markdownlint.yaml`
        ed.test(this.fwFolderUri.fsPath) && (t = t.replace(td, "/")), 
    // UNC authority reattached
    // `/<user>/dotfiles/.markdownlint.yaml` to `file://<server>/<user>/dotfiles/.markdownlint.yaml`
        this.fwFolderUri.with({path: t})
}
```

The exact error output is:

```plain
[23:52:12] ERROR: Exception while linting with markdownlint-cli2:
Error: Unable to use configuration file 'C:/Users/<user>/dotfiles/.markdownlint.yaml'; Error: UNKNOWN: unknown error, open '\\<server>\<user>\dotfiles\.markdownlint.yaml'
stack:
 at Pf (c:\Users\<user>\.vscode\extensions\davidanson.vscode-markdownlint-0.62.1\bundle.js:262:42352)
 at Uf (c:\Users\<user>\.vscode\extensions\davidanson.vscode-markdownlint-0.62.1\bundle.js:262:44570)
 at async tp (c:\Users\<user>\.vscode\extensions\davidanson.vscode-markdownlint-0.62.1\bundle.js:262:51430)
cause:
 Unknown (FileSystemError) (FileSystemError): Error: UNKNOWN: unknown error, open '\\<server>\<user>\dotfiles\.markdownlint.yaml'
 stack:
  at ar._handleError (file:///c:/Users/<user>/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:544:21685)
  at Object.readFile (file:///c:/Users/<user>/AppData/Local/Programs/Microsoft%20VS%20Code/df53daabb1/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:544:19809)
```

**Comment by DavidAnson** [comment]
Thank you! That function is mostly workarounds for Windows path weirdness. I'll have a look in a few days.

https://github.com/DavidAnson/vscode-markdownlint/blob/21618bcf521a277bb0d2c98e6656da8af1ab255f/extension.mjs#L164
